### PR TITLE
Add filters for dead and alive states

### PR DESCRIPTION
**Pull request by Pugzy**
Add filter for checking if a player is alive or dead.

Dynamic so that it's rise can be used to apply a kit.

Signed-off-by: Pugzy <pugzy@mail.com>

### CODE REVIEW

**Comment by Pablete1234** [comment]
Personally i believe we should use `<interacting/>` instead of "alive", otherwise someone joining obs and re-joining should technically always have been alive, but you'd still want to re-give the kit once they join.

Currently the `<alive/>` filter is going to say obs aren't alive, which can be a bit misleading (same for dead, obs are neither dead nor alive)

File: core/src/main/java/tc/oc/pgm/filters/matcher/player/PlayerStateFilter.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  private final boolean alive;
+
+  public PlayerStateFilter(boolean alive) {
+    this.alive = alive;
+  }
+
+  @Override
+  public Collection<Class<? extends Event>> getRelevantEvents() {
+    return ImmutableList.of(PlayerSpawnEvent.class, MatchPlayerDeathEvent.class);
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/filters/matcher/player/PlayerStateFilter.java]: I'd use `ParticipantSpawnEvent` and `ParticipantDespawnEvent`, they won't work for player going from dead to obs tho, which is why i'd consider just making it `<interacting/>` instead which should be fully covered by those 2 events

**Comment by Pablete1234** [comment]
@Electroid i'd like your input/opinion here, how do you feel about alive/dead naming where observers are never considered neither dead nor alive (they will always deny the query)?

I think "interacting" may be a better name for "alive and on a team" (being more direct than just "alive" where one may think obs are alive, when they wouldn't be considered alive). Regardless it's not something good documentation can't solve (making it clear that depsite it being called "alive" observers are never considered alive)

**Comment by Electroid** [comment]
@Pablete1234 Naming of alive and dead seems fine, though I would think observers would always abstain from both queries.

**Comment by Pablete1234** [comment]
Abstaining isn't an option for dynamic filters, must always allow or deny to the query as long as it's the correct type of query (ie: MatchPlayer)

**Review by Pablete1234 — APPROVED** [review]

**Review by Electroid — APPROVED** [review]

**Pull request merged**